gulpfile.ts

```diff
@@ -0,0 +1,159 @@
+import gulp from 'gulp';
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const clean = require('gulp-clean');
+import merge from 'merge-stream';
+import ts from 'gulp-typescript';
+import sourcemaps from 'gulp-sourcemaps';
+import terser from 'gulp-terser';
+import uglify from 'gulp-uglify';
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const rollup = require('gulp-rollup-2').rollup;
+import rollupTypescript from 'rollup-plugin-typescript2';
+
+import minifyPrivatesTransformer from 'ts-transformer-minify-privates';
+import tsMacroTransformer from 'ts-macros';
+import pathsTransformer from 'ts-transform-paths';
+import { CustomTransformerFactory, Program } from 'typescript';
+
+function createProject(overrideSettings?: ts.Settings) {
+  return ts.createProject('tsconfig.json', {
+    getCustomTransformers: (program?: Program) => {
+      if (program === undefined) throw new Error('Program is undefined');
+      return {
+        ...pathsTransformer(program),
+        before: [
+          tsMacroTransformer(program) as unknown as CustomTransformerFactory,
+          minifyPrivatesTransformer(program)
+        ]
+      };
+    },
+    ...overrideSettings
+  });
+}
+
+function terserStream() {
+  return terser({
+    compress: false,
+    output: {
+      beautify: true
+    },
+    mangle: {
+      keep_fnames: true,
```

Comment:
Why keep functions name here?

Reply:
After terser, mangle the function name through uglifyjs.

I did so because the compression rate of uglifyjs is higher